docs/contributing-to-registry.md

```diff
@@ -0,0 +1,66 @@
+>**Note**: Please ignore the doc for now as it's a work in progress. We are still working on creating the Bicep public registry, and the tool mentioned in the doc has not been published yet.
+
+# Contributing to Bicep public registry
+
+Currently, we only accept contributions from Microsoft employees. The guide was created to help teams within Microsoft with the development of Bicep public registry modules. External customers can propose a new module by opening an [issue]() in the [Azure/azure-quickstart-templates](https://github.com/Azure/azure-quickstart-templates/tree/master/modules) repository.
+
+## Prerequisite
+- Create a fork of the [Azure/azure-quickstart-templates](https://github.com/Azure/azure-quickstart-templates/tree/master/modules) repository and clone the fork to your local machine.
+<!-- TODO: Add link to Nuget once the tool is published -->
+- Install the [bicep-registry-module]() .NET tool by running:
+  - `dotnet tool install -g bicep-registry-module`
+
+## Creating a new module
+
+### Making a proposal for new module
+<!-- TODO: create an issue template in the quickstart repo -->
+Before creating a new module, you must fill out this [issue template](https://github.com/Azure/azure-quickstart-templates/issues/new) to make a proposal. Once the proposal is approved, proceed with the following steps.
+
+### Creating a directory for the new module
+<!-- TODO: need to discuss the pattern of the module path -->
+Add a new directory under the `modules` folder in your local azure-quickstart-templates repository with the path following the pattern `<ResourceProviderName>/<ModuleName>/<ModuleMajorVersion>.<ModuleMinorVersion>`. For examples: 
+- `Microsoft.Storage/storageAccounts/0.9`
```

Comment:
I think we decided to enforce lower case convention for directory names to avoid mixed casing issues when the repo is cloned on Windows. This should also be enforced by the validation.

Reply:
Good catch. I'll send a different PR to implement the validation in the `validate` command (together with the module name change).

Reply:
Updated the examples to use lowercase btw.